Core/OrderHelper.php

```diff
@@ -429,4 +471,44 @@ private static function _getLastOrderIdFromDb($sUserId)
         $sQuery = "SELECT `OXID` from oxorder WHERE `OXUSERID`=? ORDER BY `OXORDERDATE` DESC LIMIT 1";
         return DatabaseProvider::getDb()->getOne($sQuery, [$sUserId]);
     }
+
+    /**
+     * Get the current shipping address selected for the order
+     *
+     * @return array
+     *
+     * @since 1.3.0
+     */
+    public static function getSelectedShippingAddress()
+    {
+        $oOrder = oxNew(Order::class);
+        $oCurrentAddress =  $oOrder->getDelAddressInfo();
+
+        if (!is_null($oCurrentAddress)) {
```

Comment:
Also this is not very clean. Again the same parameters are returned  just the parameter is named differently. You can split it up into a separate method where the data is filled up you pass into it the $oCurrentAddress or the $oUser depending on an if.

Reply:
here the first one is from the address db table, the second from the user db table

Reply:
Here i see the same issue as with the method above.

Reply:
see answer: https://github.com/wirecard/oxid-ee/pull/174/files/262fcf6670047123b8361756463fbd63c1e4c598#r307258277

However, we would also suggest to split it into **2 functions** to make the difference more obvious (`getCurrentDeliveryAddress` & `getUserAddress`).